lib/service-data/service-data.js

```diff
@@ -512,8 +512,74 @@ external.setInstanceProperty = (id, property, value) => {
     })
   }
   const instance = external.getDiscreteInstance(id)
-  const newInstance = external.getSourceInstance(id, instance.$source)
+  let newInstance = external.getSourceInstance(id, instance.$source)
+  const oldProp = newInstance[property]
   set(newInstance, property, value)
+  if (property === 'name' || property === 'value') {
+    const updatePropertyReferences = (propName, query, applyFn) => {
+      const allInstances = external.getServiceInstances()
+      const updateIdentifier = (instId) => {
+        let inst = instId
+        if (typeof inst === 'string') {
+          inst = allInstances[instId]
+          inst = external.getSourceInstance(instId, inst.$source)
+        }
+        jp.apply(inst, query, applyFn)
+        return inst
+      }
+      newInstance = updateIdentifier(newInstance)
+      const idsWithRefs = Object.keys(allInstances).filter(instId => {
+        if (instId === 'sourceInstances') {
+          return
+        }
+        if (instId === newInstance._id) {
+          return
+        }
+        let inst = allInstances[instId]
+        inst = external.getSourceInstance(instId, inst.$source)
+        return jp.query(inst, query).length
+      }).map(instId => updateIdentifier(instId))
+
+      if (idsWithRefs.length) {
+        const updatePromises = Promise.all(idsWithRefs.map(inst => external.setInstance(inst)))
+        return updatePromises
```

Comment:
Is it intentional that this code is _not_ blocking?

Reply:
It is intentional